subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -2122,6 +2479,24 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 		     cmd_wifi_set_rts_threshold,
 		     2,
 		     0),
+	SHELL_CMD_ARG(dpp,
+		      NULL,
+		      "DPP actions refer to hostap\n"
+		      "dpp_configurator_add [curve=P-256] [net_access_key_curve=P-256]\n"
+		      "dpp_auth_init [peer=<peer_bi_id>] [conf=sta-dpp/ap-dpp]"
+		      " [ssid=<SSID>] [configurator=<configurator_id>]"
+		      " [role=<configurator/enrollee/either>]"
+		      "dpp_qr_code <qr_code>\n"
+		      "dpp_chirp own=<own_bi_id> listen=<freq>\n"
+		      "dpp_listen <freq> role=<configurator/enrollee/either>\n"
+		      "dpp_bootstrap_gen type=qrcode chan=<opclass/channel> mac=<mac addr>"
+		      " [curve=P-256]\n"
+		      "set dpp_configurator_params [peer=peer_bi_id] [conf=sta-dpp] [ssid=<SSID>]"
+		      " [configurator=1] [curve=P-256] [net_access_key_curve=P-256]\n"
+		      "set dpp_resp_wait_time <time_ms>\n",
```

Comment:
for new command we shifted to usign `getopt` as it makes parsing and entering commands easier.

Reply:
DPP commands are simply tight to wpa_cli format.
That's easier for user to switch from linux wpa_supplicant to zephyr wpa_supplicant.

Reply:
so is this okay this way for DPP?

Reply:
Sure, but in general we are not maintaining compatibility of wpa_cli vs wifi shell, for non-user cases like Certifications you can use `wpa_cli` directly https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hostap/src/wpa_cli.c#L17 and native Zephyr commands are aligned with Zephyr usage.

E.g., a typical DPP usage in Zephyr may not need all these low-level commands, so, you can add an abstraction to limit/document these, so, `wifi shell` is more controlled and `wpa_cli` takes any supported command.

Reply:
yes this dose ring a bell. I can restructure dpp shell to a more reasonable way by args with '-' and in subcommands,
which may be easier to extend.
I'll change this commit ASAP.

Reply:
updated wifi_shell.c as this comment requierd